scene/animation/animation_blend_tree.cpp

```diff
@@ -696,6 +696,10 @@ AnimationNode::NodeTimeInfo AnimationNodeOneShot::_process(const AnimationMixer:
 	set_parameter(fade_in_remaining, cur_fade_in_remaining);
 	set_parameter(fade_out_remaining, cur_fade_out_remaining);
 
+	if (cur_active != (bool)get_parameter(active)) {
+		_notify_tree(cur_active ? NOTIFY_FINISHED : NOTIFY_STARTED);
+	}
+
```

Comment:
Since the states of internal_active and current_active are not synchronized, it might be better to have an explanation in the documentation that they are “visual” and different from the "internal state".

To give a detailed example, in the case of trying to manage Node state with user scripts, OneShot has been internally terminated already when fading out, so restarting AnimationTree should cause a synchronization discrepancy if you are relying on signals.

It may be possible to add a signal STATE_CHANGED as well as documentation to solve this.

Reply:
I've added `NOTIFICATION_FADEOUT_STARTED` to allow this to be tracked by a user. Currently only one shot nodes emit it since state machines only have crossfade on their internal nodes and that can be fully tracked by the node started/finished events. It also only emits specifically when there is non-zero fadeout, so a bit different from if it was strictly a version of this line but for internal_active.